source/vistas/ui/windows/linreg_dialog.py

```diff
@@ -11,54 +11,374 @@
 mpl.use('WXAgg')
 import matplotlib.backends.backend_wxagg as wxagg
 
+#TO DRAW RECTANGLE
+from matplotlib.patches import Rectangle
+
 from vistas.ui.project import Project
 from vistas.core.timeline import Timeline
 from vistas.ui.utils import get_main_window
 from vistas.ui.events import EVT_TIMELINE_CHANGED
 
-
 class LinRegDialog(wx.Frame):
     def __init__(self, parent=None):
         super().__init__(parent, title='Linear Regression', size=(800,800))
+
+        # GLOBAL VARIABLES
+        self.zoom_mode = 0 #0 is off, 1 allows user to draw box on graph
```

Comment:
Since this has two states, a boolean would be better here. And for the sake of clarity, should be named according to what it controls, such as `can_zoom` or `zoom_box_enabled`.

```python
self.can_zoom = True
```